packages/base/src/layerBrowser/layerBrowserDialog.tsx

```diff
@@ -183,20 +278,52 @@ export const LayerBrowserComponent = ({
   );
 };
 
-export class LayerBrowserWidget extends ReactWidget {
-  private _model: IJupyterGISModel;
-  private _registry: IRasterLayerGalleryEntry[];
+export class LayerBrowserWidget extends Dialog<boolean> {
+  constructor(
+    model: IJupyterGISModel,
+    registry: IRasterLayerGalleryEntry[],
+    formSchemaRegistry: IJGISFormSchemaRegistry
+  ) {
+    // This is a bit complex so that "this" is defined
+    let _disposeDialog: (result: boolean) => void;
+    let _setDisposeDialog: () => void;
+    const _onDisposeSet = new Promise<void>(resolve => {
+      _setDisposeDialog = resolve;
+    });
+    const onDispose = new Promise<boolean>(resolve => {
+      _disposeDialog = resolve;
+      _setDisposeDialog();
+    });
```

Comment:
This part seems a bit complex, (I don't really understand).

I wonder if we could not use a lumino `PromiseDelegate<boolean>` instead of this.
The promise would resolve with the `accept` status, when the dialog is closed.

Reply:
Yeah I should definitely simplify this part. Had to workaround the fact that `this` was not defined due to implementing this prior to the `super` call. 

I'll look at using a PromiseDelegate 👍🏽

Reply:
Thanks, using a PromiseDelegate helped simplify it 👍🏽